service/scaladsl/kafka/server/src/main/scala/com/lightbend/lagom/scaladsl/broker/kafka/LagomKafkaComponents.scala

```diff
@@ -16,6 +16,12 @@ trait LagomKafkaComponents extends LagomKafkaClientComponents {
   def lagomServer: LagomServer
   def offsetStore: OffsetStore
 
+  override def topicPublisherName: Option[String] = super.topicPublisherName match {
+    case Some(other) =>
+      sys.error(s"Cannot provide the kafka topic factory as the default topic publisher since a default topic publisher has already been mixed into this cake: $other")
```

Comment:
OK I read some more and I guess 2 isn't really a problem because the trait can't appear multiple times in the list of supertypes.

So this is probably a mostly hypothetical problem. I'm still in favor of linking errors to doc pages, but maybe we can do that as a later improvement.